### PR TITLE
[ENH] Handle null values in concatenate_columns function

### LINKED ISSUE #756: concatenate_columns adds the function of ignoring empty values
# Problem description

When I use the function of concatenating multiple columns, I find that it does not handle null values as expected.

## This is the current output

```Python
df.concatenate_columns(["cat_1","cat_2","cat_3"],"cat",sep=",")
```

<table border="1" class="dataframe">  <thead>    <tr style="text-align: right;">      <th></th>      <th>cat_1</th>      <th>cat_2</th>      <th>cat_3</th>      <th>cat</th>    </tr>  </thead>  <tbody>    <tr>      <th>0</th>      <td>a</td>      <td>b</td>      <td>c</td>      <td>a,b,c</td>    </tr>    <tr>      <th>1</th>      <td>c</td>      <td>NaN</td>      <td>f</td>      <td>c,nan,f</td>    </tr>    <tr>      <th>2</th>      <td>a</td>      <td>b</td>      <td>NaN</td>      <td>a,b,nan</td>    </tr>  </tbody></table>

## Here is the output I want:

```Python
df.concatenate_columns(["cat_1","cat_2","cat_3"],"cat",sep=",",ignore_empty=True)
```
<table border="1" class="dataframe">  <thead>    <tr style="text-align: right;">      <th></th>      <th>cat_1</th>      <th>cat_2</th>      <th>cat_3</th>      <th>cat</th>    </tr>  </thead>  <tbody>    <tr>      <th>0</th>      <td>a</td>      <td>b</td>      <td>c</td>      <td>a,b,c</td>    </tr>    <tr>      <th>1</th>      <td>c</td>      <td>NaN</td>      <td>f</td>      <td>c,f</td>    </tr>    <tr>      <th>2</th>      <td>a</td>      <td>b</td>      <td>NaN</td>      <td>a,b</td>    </tr>  </tbody></table>

### PR DESCRIPTION
**Pull request by farhanreynaldo**
# PR Description

Please describe the changes proposed in the pull request:

- Add `ignore_empty` parameter to ignore null values during concatenating columns.
- Simplify the current implementation.

**This PR resolves https://github.com/pyjanitor-devs/pyjanitor/issues/756**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [ ] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [ ] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Review by samukweku — CHANGES_REQUESTED** [review]
@farhanreynaldo thanks for the PR. Could you kindly add tests for the enhancement.

**Comment by farhanreynaldo** [comment]
@samukweku This is my first time writing the unittest, is the following test sufficient?

```
@pytest.mark.functions
def test_concatenate_columns_null_values(missingdata_df):
    df = missingdata_df.concatenate_columns(
        column_names=["a", "decorated-elephant"],
        sep="-",
        new_column_name="index",
    )
    expected_values = ['1.0-1', '2.0-2', '3'] * 3
    assert expected_values == df['index'].tolist()
```

**Comment by samukweku** [comment]
@farhanreynaldo yea, it should, as long as the tests pass

**Review by samukweku — APPROVED** [review]

File: janitor/functions.py [hunk 1]
```diff
@@ -1402,19 +1403,26 @@ def concatenate_columns(
...
-        if i == 0:
-            df[new_column_name] = df[col].astype(str)
-        else:
-            df[new_column_name] = (
-                df[new_column_name] + sep + df[col].astype(str)
-            )
+
+    df[new_column_name] = (
+        df[column_names].fillna("").astype(str).agg(sep.join, axis=1)
+    )
```
Comment (ericmjl) [inline janitor/functions.py:1416]: Wow, that's pretty smaht. Nice work here, @farhanreynaldo!

**Review by ericmjl — APPROVED** [review]

**Comment by ericmjl** [comment]
Thank you, @farhanreynaldo! Great PR - very nice and defined unit of work :smile:, and you executed well on it!

**Pull request merged**